### ISSUE TITLE
`expose_functions()` doesn't work with return type int

### ISSUE THREAD

**Issue by gowerc** [stan-dev/cmdstanr#740]
**Describe the bug**
It appears that `expose_functions()` errors if a functions return type is int.  Simply changing the return type to real doesn't appear to have this issue.

**To Reproduce**

Stan file

```stan
functions {
    int myf(int x) {
        return x;
    }
}
data { real y_mean;}
parameters { real y; }
model { y ~normal(y_mean, 1); }
```

R file
```R
mod <- cmdstan_model("stanfile.stan")
fit <- mod$sample(data = list(y_mean = 0), refresh = 0)
fit$expose_functions()
```

Results in:

```
> fit$expose_functions()
Compiling standalone functions...
Error in struct_rtn:struct_op_start : argument of length 0
```

**Expected behavior**
That the function would compile and work as expected :) 

**Operating system**
Mac OS 12.6.1

**CmdStanR version number**

0.5.3 though I installed directly from GitHub ~4 days ago

**Additional context**
Nope

**Comment by andrjohns** [comment]
Thanks for finding and reporting this! I'll fix and add some broader tests

**Comment by andrjohns** [comment]
Closed by #742
